rq_dashboard/scripts/rq_dashboard.py

```diff
@@ -71,17 +86,35 @@ def configure_app(app, options):
         app.config['REDIS_DB'] = options.redis_database
     if options.poll_interval:
         app.config['RQ_POLL_INTERVAL'] = options.poll_interval
-    if options.url_prefix:
-        app.config['URL_PREFIX'] = options.url_prefix
-    app.config['REDIS_URL'] = options.redis_url_connection or None
 
 
-def run_app(app):
-    print('RQ Dashboard, version %s' % VERSION)
-    rq_dashboard.RQDashboard(app, url_prefix=app.config.get('URL_PREFIX', ''))
-    app.run(host=app.config['BIND_ADDR'], port=app.config['PORT'])
+def add_basic_auth(blueprint, username, password, realm='RQ Dashboard'):
+    """Add HTTP Basic Auth to the blueprint.
+
+    Note this is only for casual use!
 
+    """
+    @blueprint.before_request
+    def basic_http_auth(*args, **kwargs):
+        auth = request.authorization
+        if (
+                auth is None
+                or auth.password != password
+                or auth.username != username):
+            return Response(
+                'Please login',
+                401,
+                {'WWW-Authenticate': 'Basic realm="{}"'.format(realm)})
 
-if __name__ == '__main__':
-    main()
 
+def main():
+    """Command line entry point defined in setup.py."""
+    print('RQ Dashboard version {}'.format(get_version()))
+    blueprint = rq_dashboard.blueprint.blueprint
+    options = get_options()
+    if options.username:
+        add_basic_auth(blueprint, options.username, options.password)
```

Comment:
In other words, not just on the command line right? Hardly a good place for a password, even if it is only for casual use!

I was trying to keep separate the configuration of the blueprint (required however it is used) from the "configuration" of the `scripts/rq_dashboard.py` invocation. If we moved to http://click.pocoo.org/4/ as per #82 then retrieving settings from environment variables will be particular easy and give us that.

Does that sound like a plan to you?